### DISCUSSION TITLE
WebSocket connection to 'ws://test.com/graphql' failed ?

### DISCUSSION THREAD

**Discussion by gurupal** [vercel/next.js#18539] [Help]
I there any solution for this ?

WebSocket connection to 'ws://test.com/graphql' failed: Error during WebSocket handshake: Unexpected response code: 301

`const createWSLink = () => {
  return new WebSocketLink(
    new SubscriptionClient('ws://test.com/graphql', {
      lazy: true,
      reconnect: true,
      connectionParams: async () => {
        await requestAccessToken() // happens on the client
        return {
          headers: {
            authorization: accessToken ? `Bearer ${accessToken}` : '',
          },
        }
      },
    })
  )
}`

**Comment by Gideon877** [comment]
@gurupal I have the same issue. Did you manage to get this work?
